src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -178,6 +179,16 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
             {
                 try
                 {
+                    // FeatureManagement assemblies may not be loaded on provider startup, so version information is gathered upon first refresh
+                    if (!_featureManagementVersionsTelemetrySet)
```

Comment:
I'm thinking of organizing this a bit differently so it doesn't interrupt the refresh flow as much.

a private method `EnsureTelemetry` and the flag is changed to `_isTelemetrySet`

So the addition here just becomes `EnsureTelemetry()`;

Reply:
Pulling it out to the private method would definitely make it cleaner, but since we already have other telemetry (like feature filters) is it unclear to just call the method `EnsureTelemetry`?

Reply:
Doesn't seem unclear to me.